### DISCUSSION TITLE
Real-time audio analysis using SLIM models.

### DISCUSSION THREAD

**Discussion by lubomirovna** [llmware-ai/llmware#557] [Q&A] [answered]
Hello! I would like to use the slim models for real-time audio analysis. What execution method could be the most suitable for that purpose, using LLMfx class or Model catalog? Is there a difference in execution time between them? Or some implementation details that could be significant for the real-time analysis, like caching? 
I would greatly appreciate your help!

**Comment by doberst** [comment, marked as answer]
@lubomirovna - this sounds awesome - and very much in-line with our thinking too   Great question on whether to use LLMfx or Model function call directly - to some degree, it will be a matter of your preference.   The LLMfx provides some nice state mechanisms for aggregating outputs as well as abstractions when dealing with multiple models concurrently, but in principle, either approach should work.   Could you keep us posted on this project?   If it turns out that you need a little more capability in either path, let us know, and we will prioritize building out those features ....

**Comment by lubomirovna** [comment]
@doberst Thank you for your thorough response! I'll experiment with both LLMfx and Model function calls to see which aligns better with our requirements. Your offer to prioritize additional features based on our needs is greatly appreciated! I'll keep you posted on our progress.
